src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -487,9 +488,13 @@ public bool WriteIndented
             {
                 VerifyMutable();
                 _referenceHandler = value;
+                ReferenceHandlingStrategy = value?.HandlingStrategy ?? ReferenceHandlingStrategy.None;
             }
         }
 
+        // The cached value used to determine if ReferenceHandler should use Preserve or IgnoreCycle semanitcs or None of them.
+        internal ReferenceHandlingStrategy ReferenceHandlingStrategy = ReferenceHandlingStrategy.None;
```

Comment:
nit: can we make this `nullable`, and have the default value of `null` indicate that no strategy is used? (That's what we do to track number handling behavior)

```suggestion
        internal ReferenceHandlingStrategy? ReferenceHandlingStrategy;
```

Reply:
I don't see number handling as nullable in JsonSerilizerOptions, but it is nullable in JsonPropertyInfo. 
Making ReferenceHandlingStrategy nullable will add one extra null check on each place where we already check for its value, not sure if its worth doing it.